server/column_scope.cc

```diff
@@ -145,6 +145,52 @@ static CellValue extractVal(const CellValue & val, CellValue *)
     return val;
 }
 
+namespace {
+    // Made to init and hold a non POD array of dynamic size in a memory safe
+    // way.
+    template<typename T, size_t MAX_STACK_ENTRIES=4096/sizeof(T)>
```

Comment:
I believe this will cause an overrun if the objects do not fall on the 4K boundary.  Can you try with an object of say 96 bytes?